includes/GbWeiss.php

```diff
@@ -231,7 +239,39 @@ public function wooCommerceOrderStatusChanged(int $orderId, string $from, string
             return;
         }
 
-        // Send request.
+        $this->updateAuthToken();
+        $this->createLogisticsOrderAndUpdateOrderState($order);
+    }
+
+    /**
+     * Creates a logistics order using the Gebrueder Weiss API and updates the status of the WooCommerce order.
+     *
+     * @param object $order The WooCommerce order.
+     * @return void
+     */
+    public function createLogisticsOrderAndUpdateOrderState(object $order)
+    {
+        $logisticsOrder = LogisticsOrderFactory::buildFromWooCommerceOrder($order);
+        $authToken = $this->settingsRepository->getAccessToken();
+        $this->writeApiClient->getConfig()->setAccessToken($authToken);
+
+        try {
+            $this->writeApiClient->logisticsOrderPost($logisticsOrder);
+            $order->set_status("on-hold");
+            $order->save();
+        } catch (ApiException $exception) {
+            if ($exception->getCode() === 400) {
+                // handle faulty parameters.
+                return;
+            }
+
+            if ($exception->getCode() === 409) {
+                // handle conflict.
+                return;
+            }
+
+            // retry request.
```

Comment:
is this part of another ticket?

Reply:
Yes there is a story for retryng the failed requests in the next sprint.